features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/ImageWireframeHelper.kt

```diff
@@ -65,7 +76,179 @@ internal class ImageWireframeHelper(
+        startIndex: Int
+    ): MutableList<MobileSegment.Wireframe> {
+        val result = mutableListOf<MobileSegment.Wireframe>()
+        var currentIndex = startIndex
+
+        view.compoundDrawables.indices.forEach {
+            val drawable = view.compoundDrawables[it]
+            val density = mappingContext.systemInformation.screenDensity
+
+            if (drawable != null) {
+                val compoundDrawableLocationMap = createCompoundDrawableLocationMap(
+                    view = view,
+                    drawable = drawable,
+                    density = density,
+                    index = it
+                )
+
+                val drawableX = compoundDrawableLocationMap[it]?.first
+                val drawableY = compoundDrawableLocationMap[it]?.second
+
+                if (drawableX != null && drawableY != null) {
+                    @Suppress("ThreadSafety") // TODO REPLAY-1861 caller thread of .map is unknown?
+                    createImageWireframe(
+                        view = view,
+                        index = ++currentIndex,
+                        x = drawableX,
+                        y = drawableY,
+                        width = drawable.intrinsicWidth
+                            .densityNormalized(density).toLong(),
+                        height = drawable.intrinsicHeight
+                            .densityNormalized(density).toLong(),
+                        drawable = drawable,
+                        shapeStyle = null,
+                        border = null
+                    )?.let { resultWireframe ->
+                        result.add(resultWireframe)
+                    }
+                }
+            }
+        }
+
+        return result
+    }
+
+    private fun createCompoundDrawableLocationMap(
+        view: TextView,
+        drawable: Drawable,
+        density: Float,
+        index: Int
+    ): HashMap<Int, Pair<Long, Long>> {
+        val compoundDrawableLocationMap = HashMap<Int, Pair<Long, Long>>()
+
+        val globalBounds = viewUtilsInternal.resolveDrawableBounds(
+            view,
+            drawable,
+            density
+        )
+
+        val drawableWidth = drawable.intrinsicWidth
+            .densityNormalized(density)
+        val drawableHeight = drawable.intrinsicHeight
+            .densityNormalized(density)
+        val viewHeight = view.height
+            .densityNormalized(density)
+        val viewWidth = view.width
+            .densityNormalized(density)
+        val viewPaddingStart = view.paddingStart
+            .densityNormalized(density)
+        val viewPaddingTop = view.paddingTop
+            .densityNormalized(density)
+        val viewPaddingBottom = view.paddingBottom
+            .densityNormalized(density)
+        val viewPaddingEnd = view.paddingEnd
+            .densityNormalized(density)
+
+        @Suppress("MagicNumber")
+        compoundDrawableLocationMap[index] = when (index) {
```

Comment:
maybe enum would be here more descriptive than just integers? Or we can use `@IntDef` as well.

Reply:
definitely an Enum or @IntDef here, that index is very very hard to read

Reply:
I'm changing it to an enum